### DISCUSSION TITLE
关于角色身份不显示 Rolenames can only be seen by spectators?

### DISCUSSION THREAD

**Discussion by FrozenFish259** [FrozenFish259/Goose_Goose_Duck_Hack#90] [Announcements 公告]
你不能看到其他玩家的身份信息，因为服务器目前只会下发身份数据给局内玩家。目前只有中途加入游戏的旁观可以查看所有玩家身份。

You can not see other players' rolenames if you are ingame player but you can achieve that by joining game halfway (as spectator) to see all other players' rolenames.
![image](https://user-images.githubusercontent.com/49237108/211046024-8777d7c5-615b-496e-b1b8-7da1d11211ee.png)
![image](https://user-images.githubusercontent.com/49237108/211046257-1598aec9-8e57-4326-b47d-3ca097e6dc3a.png)

**Comment by FrozenFish259** [comment]
临时解决方案:
1. 启动游戏（用于中途加入房间旁观）
2. 启动辅助
3. 通过[Sandboxie沙盒](https://github.com/sandboxie-plus/Sandboxie) 启动第二个游戏（用于进行游戏）

Temporary Solution:

1. Launch game(for halfway join game to spectate)
2. Run hack
5. Launch second game with [Sandboxie](https://github.com/sandboxie-plus/Sandboxie) (for ingame playing)

**Comment by AK47donk** [comment]
太棒了

**Comment by Threemua** [comment]
> 具体第三步是怎么样的，用sandboxie开启游戏，是脱机状态

**Comment by FrozenFish259** [comment]
> > 具体第三步是怎么样的，用sandboxie开启游戏，是脱机状态

没出现过这种情况，我用加速器的话，沙盒里的游戏也是能正常连接服务器的

**Comment by pangjiea** [comment]
这部分我昨晚做过实验，有的加速器存在问题，当你使用合适的加速器时，沙盒内也可以正常联机。

**Reply by FrozenFish259** [reply]
我用的是雷神，我这边是在沙盒外面开的加速器，沙盒里面的软件也会被加速。应该和加速器品牌没有关系

**Reply by pangjiea** [reply]
雷神是可以用的，但是上号器版的uu会导致沙盒内连不上，至于正版的uu能否连上不知。ak加速器应该也可以连上

**Comment by pangjiea** [comment]
昨晚亲测身份显示有效。在沙盒内启动辅助，并且在游戏开始后加入到该房间内，以观察者视角，便可以看到所有身份

**Reply by Theotak** [reply]
为什么我开沙盒进不去房间了

**Reply by pangjiea** [reply]
1.加速器的问题，ak和雷神都可以正常使用。2.房间满了，你需要等人退出游戏断线才能链接进去

**Comment by pangjiea** [comment]
但沙盒内的游戏当开启辅助到游戏结束后，从房间退出到主菜单总会崩溃

**Comment by FrozenFish259** [comment]
> 但沙盒内的游戏当开启辅助到游戏结束后，从房间退出到主菜单总会崩溃

崩溃的问题正在定位

**Comment by pangjiea** [comment]
好的，另外我发现，死后观战时重新加载游戏辅助，可以看到本轮出刀的人，但职业不能显示。
而对于中途加入的观察者而言，开启辅助后可以正确显示这些信息

**Comment by FrozenFish259** [comment]
> 好的，另外我发现，死后观战时重新加载游戏辅助，可以看到本轮出刀的人，但职业不能显示。 而对于中途加入的观察者而言，开启辅助后可以正确显示这些信息

有意思，但是本轮是否杀过人的信息我记得是和身份信息一样，非旁观看不了，除非是本地玩家被别人杀掉

**Reply by pangjiea** [reply]
经测试，开启辅助后，断线重连后可以显示出刀人，但不能显示职业

**Reply by pangjiea** [reply]
![image](https://user-images.githubusercontent.com/75373472/212449522-9486cfa7-4605-47e8-92a8-7c6a174a4d3e.png)

**Reply by pangjiea** [reply]
我是本地玩家被刀后进行的重连，本地玩家被刀会显示一个出刀者。但是重连后每轮都可以显示所有出刀者

**Reply by pangjiea** [reply]
个人推测加速器重连时可能提供了一些全局信息，比如出刀者；可以深入挖掘一下，利用这些信息，或是实现身份推断。

**Reply by pangjiea** [reply]
据我所知，有的外挂可以通报死亡方式，如鹈鹕吃掉、被刀掉、被炸死，可以寻找这些数据，来辅助身份判断；比如识别鹈鹕，识别警长，识别带刀者

**Comment by FrozenFish259** [comment]
> 但沙盒内的游戏当开启辅助到游戏结束后，从房间退出到主菜单总会崩溃

https://github.com/Liuhaixv/Goose_Goose_Duck_Hack/releases/tag/v0.6.2.4

**Comment by CrzMan121** [comment]
请问一下作者大大，可以添加增加带刀角色的攻击距离功能吗

**Comment by w1ndys** [comment]
请问作者大大，热键没反应 ins 和del

**Reply by w1ndys** [reply]
是咋回事

**Reply by FrozenFish259** [reply]
不会吧，是不是因为你笔记本？

**Reply by w1ndys** [reply]
> 不会吧，是不是因为你笔记本？

是的，我是笔记本

**Reply by FrozenFish259** [reply]
> > 不会吧，是不是因为你笔记本？
> 
> 
> 
> 是的，我是笔记本

应该是笔记本有Fn功能键锁你没关，导致你按的不是insert和del而是对应的其他功能键

**Comment by Carolmelon** [comment]
为什么我替换ddl以后，进游戏在检测的时候提示unknown file version呢

**Comment by zhenhanbb** [comment]
现在中途加入游戏作为旁观者也看不到身份信息了 是已经修复了吗

**Reply by FrozenFish259** [reply]
疑似

**Comment by FrozenFish259** [comment]
当前游戏版本已无法通过旁观从服务器获取身份信息

**Reply by zhenhanbb** [reply]
坐等你更新反封禁

**Comment by Miyazaki122** [comment]
文件中的群都无法进入，可以重新发出群邀请吗

**Comment by Pd1021** [comment]
现在已经无法通过观战者看到身份信息了吗
